src/cairo_run.rs

```diff
@@ -19,10 +19,19 @@ use crate::{
 	hooks::HOOKS_VAR_NAME,
 };
 
+/// Execute a cairo program
+/// 
+/// A `CairoRunner` and a `VirtualMachine` will be created to execute the given `Program`.
+/// Hint and `Hooks` (if any) will be applied by the `VirtualMachine`
+/// 
+/// When no error is encoutered, returns the `CairoRunner` and `VirtualMachine`.
+/// Otherwise, returns a `CairoRunError`
+/// 
+/// This function is called by `test_single_entrypoint`
```

Comment:
wdym?

Reply:
I was wondering if adding reference to the caller will give more context.
Maybe it's not a good idea.
I can remove it.

Reply:
Ohh okay I see, 
I think you can remove the direct reference to the caller function. But it's a good idea to describe in which cases this function is called.

Feel free to add a description to explain that `cairo_run` is the last step after files have been listed and compiled. It essentially runs test functions with additional functionalities compared to cairo-rs's standard runner: hooks and additional hints.

Reply:
I have removed the direct reference but I don't think we could tell we are using additional functionalities compared to cairo-rs's standard runner since we are using cairo-rs crate.
As far as I understand we don't explicitly modify CairoRunner functionnality in cairo-foundry source code.

Reply:
True, we simply "wrap" it:
- We add our hooks to the cairo-runner instance
- We add our test hints (expect revert and mock call) to the execution scope